src/pykeen/training/training_loop.py

```diff
@@ -673,3 +745,64 @@ def _free_graph_and_cache(self):
         self.model.regularizer.reset()
         # The cache of the previous run has to be freed to allow accurate memory availability estimates
         torch.cuda.empty_cache()
+
+    def _save_state(self, path: str, stopper: Optional[Stopper] = None) -> None:
+        """Save the state of the training loop.
+
+        :param path:
+            Path of the file where to store the state in.
+        :param stopper:
+            An instance of :class:`pykeen.stopper.EarlyStopper` with settings for checking
+            if training should stop early
+        """
+        logger.debug("=> Saving checkpoint.")
+
+        if stopper is None:
+            stopper_dict = dict()
+        else:
+            stopper_dict = stopper.get_summary_dict()
+
+        torch.save(
+            {
+                'epoch': self._epoch,
+                'loss': self.losses_per_epochs,
+                'model_state_dict': self.model.state_dict(),
+                'optimizer_state_dict': self.optimizer.state_dict(),
+                'checksum': self.checksum,
+                'stopper_dict': stopper_dict,
+                'random_state': random.getstate(),
+                'np_random_state': np.random.get_state(),
+                'torch_random_state': torch.random.get_rng_state(),
```

Comment:
@cthoyt @mberr 
Considering everything relevant here - Don't we also have stateful regularizers?
And is there anything you guys can think of?

Reply:
Yes, there are e.g. the weights of the regularizer, cf. 
https://github.com/pykeen/pykeen/blob/dc247da229bf0788b526b151c98148e0e617d64e/src/pykeen/regularizers.py#L27-L85

Reply:
But, since the regularizer is a module, and its an attribute of `Model`, its pytorch weights should get saved. From [here](https://discuss.pytorch.org/t/what-is-the-difference-between-register-buffer-and-register-parameter-of-nn-module/32723) it seems that this also holds for buffers.